### PR TITLE
[master] Porting #51701 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #51701 to master

### CODE REVIEW

File: salt/modules/chocolatey.py [hunk 1]
```diff
@@ -930,8 +930,14 @@ def version(name, check_remote=False, source=None, pre_versions=False):
...
-            packages[pkg] = {'installed': installed[pkg],
-                             'available': available[pkg]}
+            # Grab the current version from the package that was installed
+            packages[pkg] = {'installed': installed[pkg]}
+
+            # If there's a remote package available, then also include that
+            # in the dictionary that we return.
+            if pkg in available:
+                packages[pkg]['available'] = available[pkg]
+            continue
```
Comment (waynew) [inline salt/modules/chocolatey.py]: No need for implicit continue

File: salt/modules/chocolatey.py [hunk 2]
```diff
@@ -930,8 +930,14 @@ def version(name, check_remote=False, source=None, pre_versions=False):
         available = list_(narrow=name, pre_versions=pre_versions, source=source)
 
         for pkg in packages:
-            packages[pkg] = {'installed': installed[pkg],
-                             'available': available[pkg]}
+            # Grab the current version from the package that was installed
+            packages[pkg] = {'installed': installed[pkg]}
```
Comment (waynew) [inline salt/modules/chocolatey.py]: I wonder - would this be better as `'available': available.get(pkg)`?

It produces an output like this:
```
local:
    ----------
    hosts:
        ----------
        available:
            None
        installed:
            - 1.1.0
```

Which feels better than:

```
local:
    ----------
    hosts:
        ----------
        installed:
            - 1.1.0
```

**Review by waynew — CHANGES_REQUESTED** [review]
Definitely needs tests, could use some updates.

**Comment by garethgreenaway** [comment]
Replaced with https://github.com/saltstack/salt/pull/56870

**Pull request closed without merging**